### ISSUE TITLE
Create a transcoding task to test Azure Batch GPU provider and fix bugs

### ISSUE THREAD

**Issue by lawrencegripper** [lawrencegripper/ion#134]
Use FFMPEG with cuda support to transcode a video on Azure Batch GPU instance to validate correct behavior.

**Comment by lawrencegripper** [comment]
Currently getting the following when attempting to run with GPU acceleration (works fine without) 

```
 libavfilter     7. 22.100 /  7. 22.100\\\\n  libswscale      5.  2.100 /  5.  2.100\\\\n  libswresample   3.  2.100 /  3.  2.100\\\\n/ion/in/data/file.raw: Invalid data found when processing input\
```

**Comment by lawrencegripper** [comment]
I believe this is related to #139

**Comment by lawrencegripper** [comment]
Saw a new error related to CUDA. It looks like the `requiresgpu` flag isn't being set on the transcoder dispatcher. 

```
[AVHWDeviceContext @ 0x3ce4740] Could not dynamically load CUDA\\n[h264_cuvid @ 0x3cd87c0] Error creating a CUDA device\\ncuvid hwaccel requested for input stream #0:0, but cannot be initialized
``

**Comment by lawrencegripper** [comment]
I've fixed the missing flag so all tasks in Azure batch now have the `--runtime nvidia` flag passed correctly to their docker images. Unfortunately I still see issues with transcoding and I think this may be due to issues with the underlying ffpmeg docker image I'm using.

I tried to switch off gpu via an environment variable but this appears to highlight a bug in pod2docker which means env vars aren't passed correctly. Fix that one here: https://github.com/lawrencegripper/pod2docker/pull/10
